tests/reftests/opamroot-versions.test

```diff
@@ -836,11 +836,11 @@ The following actions will be performed:
 <><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
 -> installed i-am-package.2
 Done.
-### opam option jobs=4 | "(${OPAMROOTVERSION})" -> "current"
+### opam option jobs=4 | "[(]${OPAMROOTVERSION}[)]" -> "current"
 GSTATE                          LOAD-GLOBAL-STATE @ ${BASEDIR}/OPAM
 [WARNING] Removing global switch 'this-internal-error' as it no longer exists
-FMT_UPG                         Light config upgrade, from 2.0 to current
-This version of opam requires an update to the layout of ${BASEDIR}/OPAM from version 2.0 to version current, which can't be reverted.
+FMT_UPG                         Light config upgrade, from 2.0 to 2.1
+This version of opam requires an update to the layout of ${BASEDIR}/OPAM from version 2.0 to version 2.1, which can't be reverted.
```

Comment:
matching against `[(]2.1[)]` will not work when `2.1` is not in a parenthesis.

Reply:
of course not, it's the point

Reply:
all the replacements in the test file were already made to match more than just the version number (I guess because of similar issues ?), in some case it matches also the surrounding quotes, in some others the next token. Here it was the parens, but they were ignored since put in a regexp unescaped.

Reply:
I don’t understand your point. Why would we want an string-replace that doesn’t replace?

Reply:
oh indeed, in the hurry was too eager in my rewrite. The patch is fine for parts that were problematic, e.g. [here](https://github.com/ocaml/opam/pull/5007/files#diff-02dcea38573b45faeca1b186a45f675f54b60c70337fdbf8f15babbd75b3c1c3L253-R255) where we *do* want to match the parens, and that allows us to discriminate. But indeed down the line it misses some rewrites.

Reply:
(I just assumed too quickly that the parens in the regexp where expected as literals originally, as they don't make much sense otherwise)